### DISCUSSION TITLE
The gateway trace and the service trace are not the same

### DISCUSSION THREAD

**Discussion by phantom-Long** [apache/skywalking#12354] [Q&A]
The environment is
skywalking-oap:10.0.1
skywalking-java-agent:9.2.0
spring-cloud-starter-gateway:4.1.4
spring-cloud-dependencies:2023.0.2
spring-boot-dependencies:3.3.0

The trace i want is "user → gateway → service-a", but actually is "user → gateway" and "user → service-a". The agent gateway jar package "apm-spring-cloud-gateway-4.x-plugin-9.2.0.jar" and "apm-spring-webflux-6.x-plugin-9.2.0.jar" have been configured in the plugins folder.

**Comment by wu-sheng** [comment]
AFAIK, if you have custom filter and codes, the trace could be broken. 

You will need to continue the trace by your codes.

**Reply by phantom-Long** [reply]
What should i do if i want to continue the trace? By the way, i just log the request info in global filter, but it cannot print the traceId, segmentId and spanId, in log4j2.xml i use the %sw_ctx param

**Reply by CzyerChen** [reply]
The traceId, segmentId and spanId can be obtained manually, you can refer to https://github.com/apache/skywalking/discussions/12272

**Reply by phantom-Long** [reply]
That means it can't print automatically yet.. Okay, what do i need to do to continue the trace, any suggestions?

**Reply by wu-sheng** [reply]
You need to check whether gateway of your version works without custom things, and figure out why custom filtwr breaks it.

We don't have your codes, and can't check for you.

**Reply by phantom-Long** [reply]
Now in gateway i have no custom things, but the trace is still broken. I also found in the gateway trace has a "SpringCloudGateway/RoutingFilter" endpoint, is that effect?

**Reply by wu-sheng** [reply]
I can't tell. I am.not a user of that.

But test scenarios cover some gateway cases.
